crates/ruff_linter/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -223,6 +227,15 @@ fn is_stub(function_def: &ast::StmtFunctionDef) -> bool {
                 Expr::StringLiteral(_) | Expr::EllipsisLiteral(_)
             )
         }
+        Stmt::Raise(StmtRaise {
+            range: _,
+            exc: exception,
+            cause: _,
+        }) => exception.as_ref().is_some_and(|exc| {
+            semantic
+                .resolve_builtin_symbol(map_callable(exc))
+                .is_some_and(|name| matches!(name, "NotImplementedError" | "NotImplemented"))
```

Comment:
Changed this to use the semantic model, and allow `raise NotImplementedError` (without argument) and `raise NotImplemented`.

Reply:
Why do you allow `raise NotImplemented`?  `NotImplemented` is not an exception.  It should never be raised.  `NotImplemented` can be returned by comparison operators.

Reply:
We have separate rules to guard against `raise NotImplemented`. But if the entire function is `raise NotImplemented`, it's almost certainly intended to be `raise NotImplementedError`, right?

Reply:
I see, fair enough!